include/Link.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+
+class Link {
+public:
+    Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t distance_ns);
+
+    void UpdateLastProcessTime(std::uint64_t new_process_time_ns);
+
+    std::shared_ptr<NetworkDevice> source;
+    std::shared_ptr<NetworkDevice> destination;
+    std::uint64_t distance_ns;
+
+    // In bits per ns
+    std::uint64_t speed = 0;
```

Comment:
Wouldn't it be better to stick to a established convention and add units of measurement to the varialbe name for self-commenting purposes? For example, `speed_bps` would be a good name. That way you don't need to add a comment and everyone using your code will have a clear understanding of what the varialbe stores straightaway.

```suggestion
    std::uint64_t speed_bps = 0;
```